### DISCUSSION TITLE
Duplicate plugin behavior when moving files

### DISCUSSION THREAD

**Discussion by nsteins** [beetbox/beets#5178] [Q&A]
I'm struggling with the behavior of the `duplicates` plugin. I would like to make a copy of all the duplicate files in another directory and then delete all of the duplicates (alternatively, I could just move them). However, when I run the `beets duplicate -k artist -k title -k album --copy dupes`, beets correctly copies the file from `music` to `dupes`, but then updates the file path of the duplicate to the file in the `dupes` folder. So now I have two files in the `music` folder, and one in the `dupes` folder, and two entries in the db, one pointing to the `music` folder and one pointing to `dupes`. So if I run `beets duplicate -k artist -k title -k album --delete`, the file in the `dupes` folder is deleted, leaving me with the same duplicate files in the `music` folder, but only one entry in the db.

Any suggestions on the way to actually accomplish my goal here? (delete duplicates but make a backup first) Also, how can I best import the copied files in `music` which are no longer in the db?

**Comment by benpage2** [comment]
If you run `beet duplicates -o PATH [other options]` it will copy the duplicates to `PATH`. Although, I'm not sure if it removes them from your library database.  

You can check this by running `beet ls PATH`. If stuff shoes up, they're still in your database and you need to run 'beet rm PATH`. For the sake of clarity this won't delete your backups but just remove them from the database.

Then run `beet duplicates -d [other options]` to delete the duplicates form your library and disk, but keep your backups at `PATH`

**Reply by nsteins** [reply]
Thank you for your help, but the issue is that there are now three copies of the file (2 in the library folder, 1 in the dupes folder), but only 2 entries in the db. So after I remove the copied duplicates using `beet rm /dupes`,  running `beet duplicates` no longer turns up duplicates because the original duplicates are not in the db.

**Reply by benpage2** [reply]
Apologies, I got a bit confused. Just to clarify, you want to use `duplicates` to delete duplicate tracks, but keep a copy of the ones that were deleted just in case there was a mistake?

**Reply by nsteins** [reply]
yes, that is ultimately my goal. I would like a back up copy of the files, but the duplicates removed from my library folder and from the beets database.

**Reply by benpage2** [reply]
If you haven't you should add the old tracks back to your library. Then the following should work for you. 

If you want to move of **all copies** of the duplicate tracks (i.e. if you have two files of the same song, move them both), run the following:

`beet duplicates -F -m PATH -k artist -k title -k album` 
- `-F` tells it to act on all copies of the duplicated track
- `-m` will move the tracks from your library to PATH
- I've included your keys as is

Then `beet remove PATH` to remove the entries for all the items in PATH from your database.


Alternatively, If you want duplicates to leave one copy of the duplicated item in your library, just remove the `-F` flag:

`beet duplicates -m PATH -k artist -k title -k album` 
- You can control which version it chooses to leave in your library by adding to the `tiebreak:` option in to config ([see the docs](https://beets.readthedocs.io/en/v1.6.0/plugins/duplicates.html))

Then same as above, run `beet remove PATH`

Let me know if something doesn't work right, I'll keep watching my inbox.

**Reply by nado** [reply]
I also got confused with this unintuitive meaning. I get the logic of it mirroring beet move/copy/delete, but only after having actually tested the different results on a test beet db.
Could we add a `beet duplicates --export PATH` which would mimic `beet move -e` and add a mention of the library update in case of a `--copy` or `--move`?
